### ISSUE TITLE
The "Funny" i18n languages could be doing more harm than good

### ISSUE THREAD

**Issue by rvedotrc** [cucumber/cucumber#809]
The en-Scouse, en-au and en-tx languages could be construed as exploiting stereotypes for comic effect, and thus could offend and alienate some users, and generally make the product look bad and less welcoming.

I'm not arguing that cucumber shouldn't support multiple languages, nor am I arguing that the languages should not be customisable.

I'm arguing that the presence of those languages could well have a negative affect – more negative than positive.

**Comment by mattwynne** [comment]
Thanks for raising this Rachel, it's not something that had ever occurred to me before.

I'd love to hear some opinions about this issue.

**Comment by duncan-bayne** [comment]
> could be construed as exploiting stereotypes for comic effect

Strewth mate, don't spit the dummy!  I don't know about them Scousers, but us true blue Aussies think it's a bit of a dag.

(Seriously ... I'm Australian, and I have no problem whatsoever with this.  The very fact that there's an en-au language in the software in the first place suggests an awareness of Australia's existence.  That's one up on many projects :smile: ).

**Comment by tigertoes** [comment]
Yeah, nah, I don't have a problem with this and I don't see how this might alienate or discourage other users. `en-au` does not feature any real stereotypical phrases and uses informal language that an [Australian English](http://en.wikipedia.org/wiki/Variation_in_Australian_English) speaker might use in everyday conversation. Even if it did somehow manage to fit in some reference to "shrimps on the barbie", "crikey" or such I'd like to think that if anyone really cared enough about it, they would simply put in a pull request.

A lot of Ruby projects are created out of developers having a bit of fun, and whilst some have made some arguably [bad decisions](http://unethicalblogger.com/2011/11/13/ten-poorly-chosen-gem-names.html), I don't believe this is one of those. Like Duncan has said, the fact someone made reference to Australia essentially shows awareness and inclusivity more than anything else.

**Comment by aslakhellesoy** [comment]
Some history:

The [Australian translation](https://github.com/cucumber/gherkin/blob/7aee3f1394c110bf216dc58712c02689aad6defd/lib/gherkin/i18n.json#L156-L168) comes from my Australian friend @delitescere (cucumber/cucumber@df07d38f6d448c75a34593157e1e9fafa375a929). Sorry Josh - I forgot to credit you.

The [Scouse](https://github.com/cucumber/gherkin/blob/7aee3f1394c110bf216dc58712c02689aad6defd/lib/gherkin/i18n.json#L212-L224) one is from Alexandra Schladebeck (then Alexandra Imrie) (cucumber/gherkin@55810dc7f5df9fc94895d750332939804d1780c7). I don't recall if she's a native speaker of the Scouse dialect, but since she gave me the translation I assume she's at least familiar with it.

The Texan one is from me, and I'm definitely not a speaker of Texan English.

The Australian and Scouse translations are so far-fetched that I'm surprise you'd label them as linguistic stereotypes. That said, if it truly offends you we'll remove them.

The Texan translation is different - I have no problem understanding how that can be seen as a linguistic stereotype. I'm sorry if I hurt anyone's feelings.

We don't plan on making any more releases of Gherkin 2 - our efforts are now on Gherkin 3. Are there any objections if we remove those the three translations in question from Gherkin 3 when that comes out?

**Comment by mattwynne** [comment]
@rvedotrc if I understand your point correctly, it doesn't really matter whether individual people from Australia, Liverpool or Texas are offended by these languages, but that someone seeing them could get the impression that this is a community who think it's funny to mock stereotypes. Is that right?

**Comment by mattwynne** [comment]
For example, if there were an `en-gay` or `en-ebonics` I would be uncomfortable with that, despite being neither gay or black. I think @rvedotrc is pointing out that these examples are at the shallow end of that curve.

**Comment by richarda** [comment]
Just recently, I was using the `en-lol` - [LOLCAT](https://github.com/cucumber/gherkin/blob/7aee3f1394c110bf216dc58712c02689aad6defd/lib/gherkin/i18n.json#L170) as a great example of how to make i18n functionality testable - you can test out i81n functionality even if you only speak English.  The extra language is also a great example of how easy it is to add another one.

I can understand the point that people might be offended or uncomfortable if they thought the tool (or tool maintainers) was not being respectful of someone else.  LOLCAT seems on the right side of the line though right?

I personally think LOLCAT is funny, and I'm a native Texan!

**Comment by rvedotrc** [comment]
I think @mattwynne and @richarda have it perfectly.

By including au/scouse/tx, a reader (and not necessarily an au/scouse/tx reader) could come away with the impression that someone is being mocked, and by extension that the project condones that. As @richarda points out, ```en-lol``` is unlikely to convey that impression.

By including au/scouse/tx, the project therefore risks being seen as a place where people might be treated less than respectfully, and the product risks being seen less professional than it would otherwise be.

_Even if the project decides to carry that risk_ by not removing these translations, the effect is wider than just this project. The tech industry benefits from being seen as a respectful place, and small things like this contribute to how the industry is seen.

**Comment by duncan-bayne** [comment]
@rvedotrc - consider also the image that the industry portrays by worrying about trivial matters like this. 

I've worked for a University and remember the atmosphere of self censorship where folks were afraid to say anything that might offend anyone on behalf of someone else. 

I agree that tech has respect and safety issues, but this sort of thing won't address them, and will likely make the culture worse.

**Comment by delitescere** [comment]
Firstly, stereotypes aren't "exploited" for comic effect, they are simply "used". Stereotypes are exploited for other purposes but we needn't get into that.

Secondly, as an Australian, I am puzzled as to why that anyone thinks my expressions aren't good enough to be used in a translation, humourous or otherwise.

Thirdly, this industry dies have problems with exclusivity and as a person in the privileged position, I shall consent to the removal of stereotypical translations, including my own.

**Comment by alexschl** [comment]
Here are my comments as the contributor of the en-Scouse translation. 

Most importantly, if there is actual offense taken - remove them. It's not that important. I had forgotten I'd contributed this. 

On being reminded of this I had the same thoughts as I did back then - it's cute and quirky and I don't see a problem with it. Most specifically because:
- In my opinion, dialects are something that should be treasured, not stifled. Language is variable and "alive" and its differences should be celebrated. 
- Any stigma attached to a dialect or accent is not inherent to the dialect - but is purely socially motivated. At least for the Scouse, one could argue that we're doing our bit to make it less stigmatized  ;) (I can't put into words enough how much that last sentence is meant both reasonably seriously and also tongue-in-cheek). 
- I can only really speak for Scouse since I am a native speaker of it and did my Masters dissertation on aspects of the dialect: the translation is probably not 100% accurate but is fair, based on the common variables used there (pluralisation of "you" done as "youse", variablility of [t] sounds with [r] or nothing for example). 
- Again, at least for Scouse, there is a fair amount of literature on its variables as well as books like "learn yerself Scouse" (I think books like this exist for most dialects). See point one about celebration of differences.

So, that's my 2cts. I also agree with @delitescere's comments above.

**Comment by aslakhellesoy** [comment]
We have established that the Scouse and Australian translations were written by native speakers who are proud of their dialect. Their motivation was to celebrate their dialect.

If anyone is mocking or offending in this matter it must be those who don't take these dialects or translations seriously. It's beyond the scope of this project to make people more tolerant, except for celebrating linguistic diversity.

The Scouse and Australian translations will remain. I have removed Texan (cucumber/gherkin@0e9322ae9e71c54590b062e4bedc812fd4c3b58c), but I'd welcome a new translation by a native speaker of that dialect.

Aslak
